tests/Pgvector.Tests/EntityFrameworkCoreTests.cs

```diff
@@ -36,20 +36,104 @@ public class Item
 
 public class EntityFrameworkCoreTests
 {
-    [Fact]
-    public async Task Main()
+    private async Task<ItemContext> GetContextAsync()
     {
-        await using var ctx = new ItemContext();
+        var ctx = new ItemContext();
         await ctx.Database.EnsureDeletedAsync();
         await ctx.Database.EnsureCreatedAsync();
+        return ctx;
+    }
+
+    [Fact]
+    public async Task Main()
+    {
+        await using var ctx = await GetContextAsync();
 
         ctx.Items.Add(new Item { Embedding = new Vector(new float[] { 1, 1, 1 }) });
         ctx.Items.Add(new Item { Embedding = new Vector(new float[] { 2, 2, 2 }) });
         ctx.Items.Add(new Item { Embedding = new Vector(new float[] { 1, 1, 2 }) });
         ctx.SaveChanges();
 
         var embedding = new Vector(new float[] { 1, 1, 1 });
-        var items = await ctx.Items.FromSql($"SELECT * FROM efcore_items ORDER BY embedding <-> {embedding} LIMIT 5").ToListAsync();
+        var items = await ctx.Items
+            .FromSql($"SELECT * FROM efcore_items ORDER BY embedding <-> {embedding} LIMIT 5")
+            .ToListAsync();
+
+        foreach (Item item in items)
+        {
+            if (item.Embedding != null)
+            {
+                Console.WriteLine(item.Embedding);
+            }
+        }
+    }
+
+    [Fact]
+    public async Task EuclideanDistance()
+    {
+        await using var ctx = await GetContextAsync();
+
+        ctx.Items.Add(new Item { Embedding = new Vector(new float[] { 1, 1, 1 }) });
```

Comment:
This probably doesn't need to be repeated for every test. As the tests don't modify the database, you can simply have an xunit class fixture which creates the database once, and then use that from all the tests.